docs/notebooks/Super Dense Coding.ipynb

```diff
@@ -0,0 +1,259 @@
+    "\n",
+    "Bob can choose between four gates to apply to the qubit. Each gate will encode a different message for Alice. The four different possibilities are listed in the table below.\n",
+    "\n",
+    "| Gates\t | Classical Message |\n",
+    "|:------:| :------: |\n",
+    "|   I\t   | 00 |\n",
+    "|   X\t   | 01 |\n",
+    "|   Z\t   | 10 |\n",
+    "|   ZX   | 11 |\n",
+    "\n",
+    "A different Bell state will be encoded for each gate that Bob applies to the qubit. After decoding, each Bell state will result in a different 2-bit message. Remember that the Identity gate doesn't alter the state, and the X and Z gates make the qubit do a pi-rotation over their respective axes.\n",
+    "\n",
+    "The different possible states after encoding will then be:\n",
+    "\n",
+    "After the encoding step, Bobs sends the qubit back to Alice. To decode the message, Alice applies a CNOT-gate followed by a H-gate, see Figure 1. The decoding of the '11' message is: H CNOT (1/sqrt(2)(-10 + 01) = 11\n",
+    "\n",
+    "The states after the different respective gate operations are listed in Table 1."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "start_time": "2024-11-20T11:02:48.197216900Z"
+    },
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def superdense_encoding_circuit(bit_a: int =0, bit_b: int =0) -> QuantumCircuit:\n",
+    "    circuit = QuantumCircuit(2, 2)\n",
+    "    qubit_a = 0\n",
+    "    qubit_b = 1\n",
+    "    #circuit.reset(qubit_a)\n",
+    "    #circuit.reset(qubit_b)\n",
+    "    circuit.measure(qubit_a, qubit_a)\n",
+    "    circuit.measure(qubit_b, qubit_b)\n",
+    "    circuit.ry(1.57079632679, qubit_a)\n",
+    "    circuit.ry(-1.57079632679, qubit_b)\n",
+    "    circuit.cz(qubit_b,qubit_a)\n",
+    "\n",
+    "    ## apply variable gate\n",
+    "    encode_gate = bit_a + 2**bit_b\n",
+    "    if encode_gate == 0: # identity I\n",
+    "        circuit.id(qubit_a)\n",
+    "    elif encode_gate == 1: # X\n",
+    "        circuit.x(qubit_a)\n",
+    "    elif encode_gate == 2: # Z\n",
+    "        circuit.z(qubit_a)\n",
+    "    elif encode_gate == 3: # Y\n",
+    "        circuit.y(qubit_a)\n",
+    "\n",
+    "    circuit.cz(qubit_b, qubit_a)\n",
+    "    circuit.ry(-1.57079632679, qubit_a)\n",
+    "    circuit.ry(1.57079632679, qubit_b)\n",
+    "\n",
+    "\n",
+    "    circuit.measure(qubit_b, qubit_a)  # in qiskit this measures in z basis\n",
+    "    circuit.measure(qubit_b, qubit_a)  # in qiskit this measures in z basis\n",
+    "    circuit.measure(qubit_b, qubit_a)  # in qiskit this measures in z basis\n",
```

Comment:
Three identical measurements?

Reply:
Their code adds three measurements for what they call maximum-likelihood 3 readout (ML3 RO), which as I have understood, is reading those three times to improve fidelity of the readout.

Reply:
Seems to me like a very backend-specific thing. But if it already was like this in the original notebooks I guess we can leave them as is.